labelbox/pagination.py

```diff
@@ -79,6 +79,32 @@ def __next__(self):
         self._data_ind += 1
         return rval
 
+    def get_one(self):
+        """Iterates over self and returns first value
+        This method is idempotent
+        """
+        for value in self:
```

Comment:
It is kind of weird how each time you call this it fetches the next page but still returns the first result.

Reply:
Maybe this could return self._data[0]

Reply:
well, technically it will fetch next page only one time. Each consecutive call won't fetch the data page:


```
datasets = client.get_datasets()

datasets.get_one() # fetches next page
datasets.get_one() # no call to server

```

Reply:
Each time you call get_one() it invokes the __next__ method which will fetch the next page and makes a call to the api.

Reply:
not exatly. I intentioally did loop here instead of next(), because it calls __iter__ firstly and sets _data_ind = 0
and the logic inside __next__ won't grab next page is previous is in memory

Reply:
oops sorry was reading one condition wrong (discussed in dms)

Reply:
do we want next_page to continue iterating until it runs out of pages?